doc/FAQ.md

```diff
@@ -119,3 +119,46 @@ we should provide it like this:
 ```java
 NO_PROXY: localhost,127.0.0.1,.google.com,.github.com
 ```
+
+### Optimistic Locking Behavior
+
+Unfortunately it's a little complicated as it depends on what operation you are doing - we'll work towards ensuring the Javadocs are as informative as possible.  Here is quick overview:
+
+- Basic mutative operations such as update and all variations of patch (patch, edit, accept, serverSideApply) that operate on a given item - are all locked to the resourceVersion on that item.  If you don't want this behavior then set the resourceVersion to null:
+
+```
+resource.accept(x -> {
+  resource.getMetadata().setResourceVersion(null);
+  resource... // some other modifications    
+});
+```
+
+When the resourceVersion is null for an update the client obtains the latest resourceVersion prior to attempting the PUT operation - in a rare circumstance this may still fail due to a concurrent modification.
+
+When the resourceVersion is null for a patch the server will always attempt to perform the patch - but of course there may be conflicts to deal with if there has been an intervening modification.
+
+**Note:** that when using informers - do not make modifications to the resources obtained from the cache - especially to the resourceVersion.
+
+**Note:** it is not recommended to use serverSideApply directly against modified existing resources - the intent is to apply only the desired state owned by the applier.  See the next topic for more.
+
+- Delete is not locked by default, you may use the applicable lockResourceVersion method if you want the delete to apply only to a specific resourceVersion
+
+- Specialized mutative operations are generally unlocked - this includes scale, rolling operations (resume, pause, restart), and others.  The rationale is that you want the narrow operation to succeed even if there has been an intervening change.  If you encounter a situation where you require these operations be locked, please raise an issue so that we can see about making the locking function applicable.
+
+- A handful of additional operations, such as undo, updateImage, and others are currently locked by default.  This may not be intentional - under the covers this is apply a json patch; older versions of json patching that created the resource diff were unlocked by default.  If you encounter an exception due to a concurrent modification performing an operation that seems like it should ignore that possibility by default please raise an issue.
```

Comment:
Late to the party but this paragraph doesn't really parse for me. The first sentence says that these operations are locked yet the subsequent sentences seem to imply that this is because we use json patch apply, which was unlocked by default. I think we should clarify this.

Reply:
Old behavior - they were implicitly unlocked
New behavior - they are locked, but that may not be intentional